sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -594,6 +607,446 @@ public CarbonWriterBuilder withJsonInput(Schema carbonSchema) {
     return this;
   }
 
+  private void validateCsvFiles() throws IOException {
```

Comment:
These validate methods(validate AvroFiles, ParquetFiles, CsvFiles, JsonFiles, OrcFiles) can be in the respective type of carbon writers. Because the way validate methods are implemented are specific to the respective format. Validate methods get readers/parse. Very similar code is in respective writers.

Also can the validation method be called from CarbonWriterBuilder.build() method based respecitve writerType if dataFiles are not null ? I think, it can be abstract method in CarbonWriter to set input files to read, that is implemented by all the writers. Writers can validate and set them.

Reply:
Moved all the validation methods to the respective writers and getting validated them from the build() method now.